### PR TITLE
Fix js message pivot selection

### PR DESCRIPTION
**Pull request by artiz**
I have found that protoc compiler incorrectly set field values in proto3 generated message wrappers in JavaScript. Incorrect implementation of jspb.Message.setFieldIgnoringDefault_ could lead to data array corruption or incorrect property value load  if field number > 500.

I have prepared simple test case at https://github.com/artiz/google-protobuf-invalid-message-pivot-selection to verify incorrect behavior

### CODE REVIEW

**Comment by rafi-kamal** [comment]
@dankurka

**Comment by artiz** [comment]
Hi @TeBoring 
Can you review this PR and plan when it could be merged into master?

**Comment by artiz** [comment]
Ping

**Comment by artiz** [comment]
Bump!

**Comment by dankurka** [comment]
This looks good to me

File: Makefile.am [hunk 1]
```diff
@@ -1217,6 +1217,7 @@ js_EXTRA_DIST=                                                         \
   js/test_bootstrap.js                                                 \
   js/testbinary.proto                                                  \
   js/testempty.proto                                                   \
+  js/testlargenumbers.proto
```
Comment (1480c1) [inline Makefile.am]: I believe this is missing a `\`

**Pull request merged**